replication-api-impl/src/main/java/org/codice/ditto/replication/api/impl/MetacardConfigLoader.java

```diff
@@ -54,97 +51,67 @@
 
   private final MetacardHelper helper;
 
-  private final MetacardType configMetacardType;
-
   private final SiteManager siteManager;
 
+  private final ReplicatorConfigManager newConfigManager;
+
+  private final Security security;
+
   public MetacardConfigLoader(
       CatalogFramework framework,
       FilterBuilder filterBuilder,
       MetacardHelper helper,
-      MetacardType configMetacardType,
-      SiteManager siteManager) {
+      SiteManager siteManager,
+      ReplicatorConfigManager newConfigManager) {
+    this(framework, filterBuilder, helper, siteManager, newConfigManager, Security.getInstance());
+  }
+
+  @VisibleForTesting
+  MetacardConfigLoader(
+      CatalogFramework framework,
+      FilterBuilder filterBuilder,
+      MetacardHelper helper,
+      SiteManager siteManager,
+      ReplicatorConfigManager newConfigManager,
+      Security security) {
     this.framework = framework;
     this.filterBuilder = filterBuilder;
     this.helper = helper;
-    this.configMetacardType = configMetacardType;
     this.siteManager = siteManager;
+    this.newConfigManager = newConfigManager;
+    this.security = security;
   }
 
-  public ReplicatorConfig createConfig() {
-    return new ReplicatorConfigImpl();
+  public void init() {
+    security.runAsAdmin(
+        () -> {
+          try {
+            security.runWithSubjectOrElevate(
+                () -> {
+                  getAllConfigs()
```

Comment:
I am wondering if we need some kind of retry logic in here for when we come up before the catalog is available.